code/core/src/common/js-package-manager/Yarn2Proxy.ts

```diff
@@ -167,13 +168,11 @@ export class Yarn2Proxy extends JsPackageManager {
       }
     }
 
-    const packageJsonPath = await findUp(
-      (dir) => {
-        const possiblePath = join(dir, 'node_modules', packageName, 'package.json');
-        return existsSync(possiblePath) ? possiblePath : undefined;
-      },
-      { cwd: basePath }
-    );
+    const dirs = walk.up(basePath ?? '.');
```

Comment:
and again
https://github.com/storybookjs/storybook/pull/31338/files#r2075527459

Reply:
yeah there's a lot of duplicated code in the proxy files 😅